### ISSUE TITLE
parsing error with temp-table in base class

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#134]
If you define a TEMP-TABLE in a class and then use it in another class that inherits the first class (in this case its an CREATE statement, i don't know if this is relevant) the following error occures:

`org.sonar.plugins.openedge.api.org.prorefactor.treeparser.TreeParserException: C:\Buildtools\Jenkins\workspace\Sonar_Runner\Some.cls:231 Could not resolve table: ttFoo`

**Comment by gquerret** [comment]
Test case added in branch Issue134: https://github.com/Riverside-Software/sonar-openedge/commit/d4fef56c4dd57bdbc59fbe9f4731648b186cffd1

@movedoa But tree parsing doesn't fail: does your issue looks like this case ?

**Comment by movedoa** [comment]
yes it look like the test case, i don't see any noticeable difference

**Comment by movedoa** [comment]
Still happens in 1.9.7
Here is the stack in case it help:

```
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TP01Support.recordNodeSymbol(TP01Support.java:870) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TP01Support.recordNameNode(TP01Support.java:919) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.tbl(TreeParser01.java:437) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.createstate(TreeParser01.java:13005) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.statement(TreeParser01.java:35301) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.blockorstate(TreeParser01.java:280) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.code_block(TreeParser01.java:11618) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.methodstate(TreeParser01.java:26587) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.statement(TreeParser01.java:34981) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.blockorstate(TreeParser01.java:280) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.code_block(TreeParser01.java:11618) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.classstate(TreeParser01.java:11565) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.statement(TreeParser01.java:34777) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.blockorstate(TreeParser01.java:280) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser01.TreeParser01.program(TreeParser01.java:120) ~[na:na]
[sonar:sonar] 	at org.sonar.plugins.openedge.api.org.prorefactor.treeparser.ParseUnit.treeParser(ParseUnit.java:164) ~[na:na]
```

**Comment by gquerret** [comment]
Rebased test case, and still can't reproduce.

**Comment by movedoa** [comment]
New Info: The error is still happening when analyzing my whole project on the server.
But the error does not happen when analyzing the class in eclipse with SonarLint.

**Comment by samixchoumi** [comment]
I've the same error on a buffer table but the strange thing is that this table work on other procedure
```
[sonar:sonar] 16:02:41.522 ERROR - Error during code parsing for lmdcall.p
[sonar:sonar] org.prorefactor.refactor.RefactorException: org.prorefactor.treeparser.TreeParserException: C:\User\workspace\BUILD46.06\temp\src\lmdcall.p:1427 Could not resolve table: b_Radio
[sonar:sonar] 	at org.prorefactor.treeparser.ParseUnit.treeParser(ParseUnit.java:178)
[sonar:sonar] 	at org.prorefactor.treeparser.ParseUnit.treeParser01(ParseUnit.java:193)
[sonar:sonar] 	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.parseMainFile(OpenEdgeProparseSensor.java:185)
[sonar:sonar] 	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.execute(OpenEdgeProparseSensor.java:136)
[sonar:sonar] 	at org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
[sonar:sonar] 	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:57)
[sonar:sonar] 	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:49)
[sonar:sonar] 	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
[sonar:sonar] 	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:175)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
[sonar:sonar] 	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:262)
[sonar:sonar] 	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:257)
[sonar:sonar] 	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:247)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
[sonar:sonar] 	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
[sonar:sonar] 	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
[sonar:sonar] 	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:118)
[sonar:sonar] 	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:117)
[sonar:sonar] 	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:77)
[sonar:sonar] 	at org.sonar.runner.batch.IsolatedLauncher.execute(IsolatedLauncher.java:48)
[sonar:sonar] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[sonar:sonar] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[sonar:sonar] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[sonar:sonar] 	at java.lang.reflect.Method.invoke(Method.java:498)
[sonar:sonar] 	at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:87)
[sonar:sonar] 	at org.sonar.runner.impl.BatchLauncher$1.run(BatchLauncher.java:75)
[sonar:sonar] 	at java.security.AccessController.doPrivileged(Native Method)
[sonar:sonar] 	at org.sonar.runner.impl.BatchLauncher.doExecute(BatchLauncher.java:69)
[sonar:sonar] 	at org.sonar.runner.impl.BatchLauncher.execute(BatchLauncher.java:50)
[sonar:sonar] 	at org.sonar.runner.api.EmbeddedRunner.doExecute(EmbeddedRunner.java:102)
[sonar:sonar] 	at org.sonar.runner.api.Runner.execute(Runner.java:100)
[sonar:sonar] 	at org.sonar.ant.SonarTask.launchAnalysis(SonarTask.java:53)
[sonar:sonar] 	at org.sonar.ant.SonarTask.execute(SonarTask.java:48)
[sonar:sonar] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
[sonar:sonar] 	at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
[sonar:sonar] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[sonar:sonar] 	at java.lang.reflect.Method.invoke(Method.java:498)
[sonar:sonar] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[sonar:sonar] 	at org.apache.tools.ant.Task.perform(Task.java:348)
[sonar:sonar] 	at org.apache.tools.ant.Target.execute(Target.java:390)
[sonar:sonar] 	at org.apache.tools.ant.Target.performTasks(Target.java:411)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
[sonar:sonar] 	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
[sonar:sonar] 	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:442)
[sonar:sonar] 	at org.apache.tools.ant.taskdefs.CallTarget.execute(CallTarget.java:105)
[sonar:sonar] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
[sonar:sonar] 	at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
[sonar:sonar] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[sonar:sonar] 	at java.lang.reflect.Method.invoke(Method.java:498)
[sonar:sonar] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[sonar:sonar] 	at org.apache.tools.ant.Task.perform(Task.java:348)
[sonar:sonar] 	at org.apache.tools.ant.Target.execute(Target.java:390)
[sonar:sonar] 	at org.apache.tools.ant.Target.performTasks(Target.java:411)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
[sonar:sonar] 	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
[sonar:sonar] 	at org.apache.tools.ant.Main.runBuild(Main.java:809)
[sonar:sonar] 	at org.apache.tools.ant.Main.startAnt(Main.java:217)
[sonar:sonar] 	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
[sonar:sonar] 	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
[sonar:sonar] Caused by: org.prorefactor.treeparser.TreeParserException: C:\User\workspace\BUILD46.06\temp\src\lmdcall.p:1427 Could not resolve table: b_Radio
[sonar:sonar] 	at org.prorefactor.treeparser01.TP01Support.recordNodeSymbol(TP01Support.java:930)
[sonar:sonar] 	at org.prorefactor.treeparser01.TP01Support.recordNameNode(TP01Support.java:979)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.tbl(TreeParser01.java:436)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.parameter(TreeParser01.java:4257)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.parameterlist_noroot(TreeParser01.java:7881)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.exprt(TreeParser01.java:4414)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.expression(TreeParser01.java:1018)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.expression(TreeParser01.java:789)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.ifstate(TreeParser01.java:39359)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.statement(TreeParser01.java:35315)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.blockorstate(TreeParser01.java:279)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.code_block(TreeParser01.java:11841)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.dostate(TreeParser01.java:23677)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.statement(TreeParser01.java:35219)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.blockorstate(TreeParser01.java:279)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.casestate(TreeParser01.java:36501)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.statement(TreeParser01.java:35123)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.blockorstate(TreeParser01.java:279)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.code_block(TreeParser01.java:11841)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.functionstate(TreeParser01.java:25145)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.statement(TreeParser01.java:35285)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.blockorstate(TreeParser01.java:279)
[sonar:sonar] 	at org.prorefactor.treeparser01.TreeParser01.program(TreeParser01.java:119)
[sonar:sonar] 	at org.prorefactor.treeparser.ParseUnit.treeParser(ParseUnit.java:176)
[sonar:sonar] 	... 65 common frames omitted
```

**Comment by gquerret** [comment]
Expected to be fixed in 2.0.4

**Comment by gquerret** [comment]
Fixed in master
